### PR TITLE
Dynamic component loading

### PR DESCRIPTION
**Pull request by fdefalco**
This PR will provide dynamic component loading to replace the current approach of toggling visibility of components.

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -64,292 +68,232 @@ define(['jquery', 'knockout', 'jnj_chart', 'd3', 'ohdsi.util', 'appConfig', 'fac
...
 						require(['data-sources'], function () {
-							self.currentView('datasources');
+							self.componentParams = {
+								model: self
+							};
+							self.currentView('data-sources');
 						});
 					},
-					'/datasources/:sourceName/:report': function (sourceName, report) {
+					'/datasources/:sourceKame/:reportName': function (sourceKey, reportName) {
```
Comment (chrisknoll) [inline js/app.js]: /:sourceKame looks incorrect.

File: js/styles/atlas.css [hunk 2]
```diff
@@ -96,12 +96,11 @@ body {
 }
 
 .heading {
```
Comment (fdefalco) [inline js/styles/atlas.css]: removed the bottom border from header elements to reduce confusion from all the horizontal lines it was introducing in some interfaces

File: js/styles/atlas.css [hunk 3]
```diff
@@ -347,17 +340,17 @@ div#wrapperMainWindowContainer {}
 
 .feFilter {
 	padding-left: 0px;
+	padding-top: 99px;
 }
 
 .fa.fa-refresh.fa-spin {
 	font-weight: normal;
 }
```
Comment (fdefalco) [inline js/styles/atlas.css]: made changes to the appearance of the facets along side components to align the facets with the table it was filtering

**Comment by chrisknoll** [comment]
@fdefalco , are these changes planned for the 2.0 release?  I'm trying to coordinate the last pre-2.0 release that incorporates fixes over the past few months, and need to know if this can sit in the 2.0 release or if there is some priority to get this change in before 2.0.

**Comment by fdefalco** [comment]
Yes, we can leave these out until the 2.0 release.

**Comment by chrisknoll** [comment]
Ok, I'm going to create a branch called dynamic-component-loading-final and have this PR squelch and merge onto that new branch.  Then when we're ready to incorporate the features for 2.0 we'll make an integration branch with all those features for pushing into the 2.0 release (or after the next release we can start folding all of this directly into master, whatever makes sense).

**Pull request merged**